### PR TITLE
feat: add update notifications

### PR DESCRIPTION
**Pull request by sinedied**
This PR adds update checking in background, performed once a day. If an update is found, a message will be shown to let the update know an update is pending.

This uses the exact same package used by NPM itself (and many other popular packages) 😉 

Note that the check is automatically disabled on CI and testing environments.

### CODE REVIEW

**Review by manekinekko — APPROVED** [review]

**Comment by anthonychu** [comment]
@sinedied  Great idea. Can you share instructions on how to test this? Thanks.

**Comment by sinedied** [comment]
@anthonychu it wasn't easy but I added a unit test for it 🙂 

If you want to test it "live":
- On a "dev" build (ie clone of this repo + `npm link`), downgrade the version in the `package.json` file to something older than the current published version
- Remove `~/.config/configstore/update-notifier-@azure/static-web-apps-cli.json` to force the package to check again for new updates (by default it's once a day)
- Run `swa -v`: it will generate the file, and wait for 1 day until the first update check
- Edit `~/.config/configstore/update-notifier-@azure/static-web-apps-cli.json` and set `lastUpdateCheck` to 0 (to force a timestamp later than a day)
- Run `swa -v`: it will check for update
- Run `swa -v` (or any other command): it will show the notification on exit

You'll see something like this:
```
   ╭────────────────────────────────────────────────────╮
   │                                                    │
   │          Update available 0.0.1 → 0.6.0            │
   │   Run npm i @azure/static-web-apps-cli to update   │
   │                                                    │
   ╰────────────────────────────────────────────────────╯
```

Note that it will automatically add `-g` to the npm command when installed globally (not with npm link).

File: src/cli/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        current: "0.1.0",
+        latest: "1.0.0",
+        name: pkg.name,
+      };
+    };
+    const orginalNotify = notifier.prototype.notify;
+    notifier.prototype.notify = function (options: any = {}) {
+      options.defer = false;
+      orginalNotify.bind(this)(options);
+    };
```
[the remark this answers is not in the document]
Comment (sinedied) [reply src/cli/index.spec.ts]: I updated the PR 😉 
I initially had typing issues with `this` doing that, but I found a workaround.

**Review by manekinekko — APPROVED** [review]

**Review by anthonychu — APPROVED** [review]
Sorry for the delay. I had to find a Windows machine to test it on as well.

**Pull request merged**